docs/source/how-to-upstream.mdx

```diff
@@ -1,28 +1,117 @@
 # Upload files to the Hub
 
-Sharing your files and work is a very important aspect of the Hub. The `huggingface_hub` uses a Git-based workflow to upload files to the Hub. You can use these functions independently or integrate them into your own library, making it more convenient for your users to interact with the Hub. This guide will show you how to:
+Sharing your files and work is an important aspect of the Hub. The `huggingface_hub` offers several options for uploading your files to the Hub. You can use these functions independently or integrate them into your library, making it more convenient for your users to interact with the Hub. This guide will show you how to:
 
-* Push files with a `commit` context manager.
+* Push files without Git.
+* Push very large files with [Git LFS](https://git-lfs.github.com/).
+* Push files with the `commit` context manager.
 * Push files with the [`~Repository.push_to_hub`] function.
-* Upload very large files with [Git LFS](https://git-lfs.github.com/).
-* Push files without Git installed with [`HfApi`]
 
 Whenever you want to upload files to the Hub, you need to log in to your Hugging Face account:
 
-1. Log in to your Hugging Face account with the following command:
+- Log in to your Hugging Face account with the following command:
 
    ```bash
    huggingface-cli login
    ```
 
-2. Alternatively, if you prefer working from a Jupyter or Colaboratory notebook, login with [`notebook_login`]:
+- Alternatively, if you prefer working from a Jupyter or Colaboratory notebook, log in with [`notebook_login`]:
 
    ```python
    >>> from huggingface_hub import notebook_login
    >>> notebook_login()
    ```
 
-   [`notebook_login`] will launch a widget in your notebook from which you can enter your Hugging Face credentials.
+   [`notebook_login`] launches a widget in your notebook from which you can enter your Hugging Face credentials.
+
+## Push files without Git
+
+If you don't have Git installed on your system, use [`create_commit`] to push your files to the Hub. [`create_commit`] uses the HTTP protocol to upload files to the Hub. Any large or binary files are automatically uploaded with Git LFS. 
```

Comment:
I would not assume people know what Git LFS is, so maybe we can remove the last sentence or move it to a tip.